### PR TITLE
US187Task245 API to return required data type of task assessment by t…

### PR DESCRIPTION
**Pull request by xiangweizheng**
…he frontend team

US187Task245 based on the communication with frontend team, make a API to return required data type  of task assessment by the frontend team. It helps to reduce API calls.

How to test:
add following codes into taiga.js
import * as taiga from '../libraries/Taiga';
 {console.log(taiga.task_assessment(2577741).then((val) => {console.log('fe', val)} ))}

Expected results:
fe {task_valid: true, finished: true, end_status: "Closed", num_stat: 3, fin_per: 100, …} detail: (3) [{…}, {…}, {…}] end_status: "Closed" fin_per: 100 finished: true num_stat: 3 task_valid: true __proto__: Object

expected detail array:
detail: Array(3) 0: {state_trans_valid: true, timecost: 0, acctime: 0, status_trans: Array(2), user: {…}, …} 1: {state_trans_valid: true, timecost: 762244801, acctime: 762244801, status_trans: Array(2), user: {…}, …} 2: {state_trans_valid: true, timecost: 1628022850, acctime: 1628022850, status_trans: Array(2), user: {…}, …}

### CODE REVIEW

File: taigit/src/libraries/Taiga.ts [hunk 1]
```diff
@@ -498,4 +498,130 @@ taiga_issues(projId : number) : Promise<Object>{
         output.push(new_entry);
     }
     return output;
-}
\ No newline at end of file
+}
+
+
```
Comment (HalcyonAura) [inline taigit/src/libraries/Taiga.ts]: Remove extra new lines, only one needed between functions

File: taigit/src/libraries/Taiga.ts [hunk 2]
```diff
@@ -498,4 +498,130 @@ taiga_issues(projId : number) : Promise<Object>{
...
+ *             The Object in the Array has the following data type:
+ *             <Object>：
+ *               state_trans_valid: true,
+ *               timecost: number,  //time of history entry cost in milliseconds
+ *               acctime: number,   //time of task accumulated of this history entry cost in milliseconds
+ *               status_trans: String[],//Status transition array
+ *               user: object,         // Taiga User Object
+ *               date: number,        //Date and time of history entry in milliseconds since epoch
+ *
+ *
```
Comment (HalcyonAura) [inline taigit/src/libraries/Taiga.ts]: Remove extra empty lines

File: taigit/src/libraries/Taiga.ts [hunk 3]
```diff
@@ -498,4 +498,130 @@ taiga_issues(projId : number) : Promise<Object>{
...
+                new_entry.state_trans_valid = false;
+                task_valid = false;
+            }
+
+        output.push(new_entry);
+        }
+    }
+
+
+
```
Comment (HalcyonAura) [inline taigit/src/libraries/Taiga.ts]: Remove extra new lines

File: taigit/src/libraries/Taiga.ts [hunk 4]
```diff
@@ -498,4 +498,130 @@ taiga_issues(projId : number) : Promise<Object>{
...
+    let t_fin_per : number;
+    let t_num_stat : number;
+    let t_finished : boolean = false;
+    if(endstatus == "Closed") {
+        t_finished = true;
+    }
+
```
Comment (HalcyonAura) [inline taigit/src/libraries/Taiga.ts]: Remove extra new lines

File: taigit/src/libraries/Taiga.ts [hunk 5]
```diff
@@ -498,4 +498,114 @@ taiga_issues(projId : number) : Promise<Object>{
...
+    let t_num_stat : number;
+    let t_finished : boolean = false;
+    if(endstatus == "Closed") {
+        t_finished = true;
+    }
+    //transfer ["New", "In progress"] ["In progress", "Ready for test"]  ["Ready for test", "Closed"])
+    //into number status
+    switch(endstatus) {
+        case "Closed": {
+            t_fin_per = 100;
```
Comment (HalcyonAura) [inline taigit/src/libraries/Taiga.ts]: Do we still need these (t_fin_per) values if all that's being returned is the transitions and whether or not the overall transition tracking was valid?

Comment (xiangweizheng) [reply taigit/src/libraries/Taiga.ts]: task finish percentage is an supplement way to show the burndown chart in task-wise accuracy.

Comment (HalcyonAura) [reply taigit/src/libraries/Taiga.ts]: Frontend from what I recall is not making a component that shows this information. @trevorforrey can elaborate/confirm. He's looking for user story history and that's it.

**Pull request merged**